crates/core/src/error.rs

```diff
@@ -367,4 +368,16 @@ impl SurfpoolError {
             "sigVerify may not be used with replaceRecentBlockhash",
         ))
     }
+
+    pub fn slot_too_old(slot: Slot) -> Self {
+        Self(Error::invalid_params(format!(
+            "Requested {slot} is before the first local slot, and no remote RPC was provided."
+        )))
+    }
+
+    pub fn get_block(e: solana_client::client_error::ClientError) -> Self {
```

Comment:
Can we improve this error to say `"Failed to get block {} from remote: {}"`, passing in the block?